Ed-Fi-Exchange-OSS/EdFi-Project-Buzz
# [FIF-185] Survey ETL Upload should delete old files

- Added new API env const: SURVEY_FILES_RETENTION_DAYS=30.
- Updated API to add a clean up task to the queue.
- Added buzzCleanUp task to the ETL.

### **Steps to Test**

- In the API you must define the retention_days variable
- You must upload a new survey file.
![CU_000](https://user-images.githubusercontent.com/56046999/89649889-b9a07100-d87e-11ea-91c2-3cc172084022.PNG)

- In the Jobs table you can see that two records are added, one to process the survey load and a cleaning request.
![CleanUp01](https://user-images.githubusercontent.com/56046999/89650064-06844780-d87f-11ea-9f22-3cdf3adee65d.PNG)

- The request to delete the file has the execution date defined according to the retention_Days parameter
- In the survey folder you can see the uploaded file.
![CU_0000](https://user-images.githubusercontent.com/56046999/89649901-befdbb80-d87e-11ea-89c3-7343dc826894.PNG)

- To test the deletion, change the execution date to today.
![CU03](https://user-images.githubusercontent.com/56046999/89650106-1c920800-d87f-11ea-826b-9cff072ce4f1.PNG)

- In the task you can see the CleanUp execution.
![CU_04](https://user-images.githubusercontent.com/56046999/89650120-2156bc00-d87f-11ea-9a0e-43636475e427.PNG)

- Check the survey folder again, the file must be deleted.
![CU_05](https://user-images.githubusercontent.com/56046999/89650127-261b7000-d87f-11ea-8d4a-d083d7a88f3a.PNG)
